src/EditorFeatures/Core/Shared/Extensions/ITextSnapshotExtensions.cs

```diff
@@ -89,5 +90,16 @@ private static IEnumerable<AbstractFormattingRule> GetFormattingRules(Document d
                 return snapshot.GetOpenDocumentInCurrentContextWithChanges();
             }
         }
+
+        /// <summary>
+        /// Get <see cref="Document"/> from <see cref="Text.Extensions.GetOpenDocumentInCurrentContextWithChanges(ITextSnapshot)"/>
+        /// once <see cref="IWorkspaceStatusService.WaitUntilFullyLoadedAsync(CancellationToken)"/> returns
+        /// </summary>
+        public static Document GetFullyLoadedOpenDocumentInCurrentContextWithChanges(
```

Comment:
I like that you have this method that internally hides the async if only to prevent people from trying to do a .Wait() without realizing it. What do we want to do with the other one that's the async method? Does it minimally need a comment that you need a JTF wait for it?

Reply:
This is an anti-pattern. The sync-to-async transition is undesirable in the implementation and should not be forced on callers, especially callers that are already asynchronous. The transition should always be moved as far to the callers as possible, and eventually eliminated. The risk of `Wait` should not exist because `Wait` is not allowed in Visual Studio. An exception is claimed for provable cases that do not require the UI thread, but this is not validated or enforced so it repeatedly leads to bugs. At a sync-to-async boundary, `Wait` should always be assumed to be a deadlock risk.